lib/membrane_h264_plugin/parser.ex

```diff
@@ -208,19 +192,91 @@ defmodule Membrane.H264.Parser do
     {[end_of_stream: :output], state}
   end
 
-  defp prepare_actions_for_aus(aus, pts, dts, state) do
-    Enum.reduce(aus, [], fn au, acc ->
-      sps_actions =
-        case Enum.find(au, &(&1.type == :sps)) do
-          nil ->
-            []
+  defp prepare_actions_for_aus(aus, state) do
+    prepare_actions_for_aus(aus, nil, nil, state)
+  end
 
-          sps_nalu ->
-            [stream_format: {:output, Format.from_sps(sps_nalu, framerate: state.framerate)}]
-        end
+  defp prepare_actions_for_aus(aus, buffer_pts, buffer_dts, state) do
+    {pts, dts} =
+      case state.mode do
+        :bytestream ->
+          if state.previous_timestamps == {nil, nil},
+            do: {-1, -1},
+            else: state.previous_timestamps
+
+        :nalu_aligned ->
+          state.previous_timestamps
+
+        :au_aligned ->
+          {buffer_pts, buffer_dts}
+      end
 
-      acc ++ sps_actions ++ [{:buffer, {:output, wrap_into_buffer(au, pts, dts)}}]
-    end)
+    {actions, au_count, profile} =
+      Enum.reduce(aus, {[], 0, state.profile}, fn au, {acc, cnt, profile} ->
+        {sps_actions, profile} = maybe_parse_sps(au, state, profile)
+        {pts, dts} = maybe_generate_timestamps(pts, dts, state, profile, cnt)
```

Comment:
hmm, you are passing `pts` as `presentation_order_number` and `dts` as `decoding_order_number` to the `maybe_generate_timestamp` function - that might be confusing. (more info in the general comment)